test_end-to-end failing

Note: these tests are now turned off, until this issue is fixed. https://github.com/oceanprotocol/df-py/commit/8a01eb3eb894d1c3ba222b42b36c23452e56d9e4

----

```text
=========================== short test summary info ============================
FAILED util/test/test_end-to-end.py::test_without_csvs - AssertionError: 6.66...
FAILED util/test/test_end-to-end.py::test_with_csvs - AssertionError: 6.66666...
```

test_without_csvs fails in two different ways:

```text
_____________________________ test_without_csvs _______________________________

    @enforce_types
    def test_without_csvs():
        chainID = networkutil.DEV_CHAINID
    
        st, fin, n = ST, FIN, 25
        rng = blockrange.BlockRange(st, fin, n)
    
        (V0, SYM0) = query.queryNftvolsAndSymbols(rng, chainID)
    
        vebals = query.queryVebalances(rng, chainID)
        allocs = query.queryAllocations(rng, chainID)
        stakes = allocations.allocsToStakes(allocs, vebals)
    
        R = {"OCEAN": 0.5, "H2O": 1.618}
    
        V, SYM = (
            {chainID: V0},
            {chainID: SYM0},
        )
    
        OCEAN_avail = 0.0001
    
        rewardsperlp, _ = calcrewards.calcRewards(stakes, V, SYM, R, OCEAN_avail)
    
        sum_ = sum(rewardsperlp[chainID].values())
>       assert sum_ == pytest.approx(OCEAN_avail, 0.01), sum_
E       AssertionError: 6.666666666666667e-05
E       assert 6.666666666666667e-05 == 0.0001 ± 1.0e-06
E        +  where 0.0001 ± 1.0e-06 = <function approx at 0x7fed1ab5d8b0>(0.0001, 0.01)
E        +    where <function approx at 0x7fed1ab5d8b0> = pytest.approx
```



```text
_________________________test_without_csvs ___________________________
    @enforce_types
    def test_without_csvs():
        chainID = networkutil.DEV_CHAINID
    
        st, fin, n = ST, FIN, 25
        rng = blockrange.BlockRange(st, fin, n)
    
>       (V0, SYM0) = query.queryNftvolsAndSymbols(rng, chainID)

util/test/test_end-to-end.py:31: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
venv/lib/python3.10/site-packages/enforce_typing/decorator.py:29: in wrapper
    return func(*args, **kwargs)
util/query.py:56: in queryNftvolsAndSymbols
    Vi_unfiltered = _queryNftvolumes(rng.st, rng.fin, chainID)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

st_block = 7957, end_block = 8012, chainID = 8996

    def _queryNftvolumes(
        st_block: int, end_block: int, chainID: int
    ) -> Dict[str, Dict[str, float]]:
        """
        @description
          Query the chain for datanft volumes within the given block range.
    
        @return
          nft_vols_at_chain -- dict of [basetoken_addr][nft_addr]:vol_amt
        """
        print("getVolumes(): begin")
    
        NFTvols: Dict[str, Dict[str, float]] = {}
    
        chunk_size = 1000  # max for subgraph = 1000
        offset = 0
        while True:
            query = """
            {
              orders(where: {block_gte:%s, block_lte:%s}, skip:%s, first:%s) {
                id,
                datatoken {
                  id
                  symbol
                  nft {
                    id
                  }
                  dispensers {
                    id
                  }
                },
                lastPriceToken,
                lastPriceValue,
                block,
                gasPrice,
                gasUsed
              }
            }
            """ % (
                st_block,
                end_block,
                offset,
                chunk_size,
            )
            offset += chunk_size
            result = submitQuery(query, chainID)
>           new_orders = result["data"]["orders"]
E           KeyError: 'data'

util/query.py:405: KeyError
```

Full log: https://github.com/oceanprotocol/df-py/actions/runs/3339771513/jobs/5529113302#step:8:2166

--- a/util/calcrewards.py
+++ b/util/calcrewards.py
@@ -119,7 +119,7 @@ def _calcRewardsUsd(S, V_USD, rewards_avail: float) -> np.ndarray:
                 stake_ij * TARGET_WPY,
             )
 
-    # filter negligible values (<0.00001% of total RF)
+    # filter negligible values
     R[R < 0.000001] = 0.0
 
     if np.sum(R) == 0.0:

--- a/util/query.py
+++ b/util/query.py
@@ -402,7 +402,10 @@ def _queryNftvolumes(
         )
         offset += chunk_size
         result = submitQuery(query, chainID)
+        if "errors" in result:
+            raise AssertionError(result)
         new_orders = result["data"]["orders"]
+
         if new_orders == []:
             break
         for order in new_orders:

--- a/util/test/test_end-to-end.py
+++ b/util/test/test_end-to-end.py
@@ -41,12 +41,12 @@ def test_without_csvs():
         {chainID: SYM0},
     )
 
-    OCEAN_avail = 0.0001
+    OCEAN_avail = 1e-6
 
     rewardsperlp, _ = calcrewards.calcRewards(stakes, V, SYM, R, OCEAN_avail)
 
     sum_ = sum(rewardsperlp[chainID].values())
-    assert sum_ == pytest.approx(OCEAN_avail, 0.01), sum_
+    assert sum_ == pytest.approx(OCEAN_avail, OCEAN_avail/1000.0), sum_
 
 
 @enforce_types
@@ -88,7 +88,7 @@ def test_with_csvs(tmp_path):
 
     stakes = allocations.loadStakes(csv_dir)  # loads allocs & vebals, then *
 
-    OCEAN_avail = 0.0001
+    OCEAN_avail = 1e-6
     rewardsperlp, _ = calcrewards.calcRewards(stakes, V, SYM, R, OCEAN_avail)
 
     sum_ = sum(rewardsperlp[chainID].values())

--- a/util/test/test_end-to-end.py
+++ b/util/test/test_end-to-end.py
@@ -92,7 +92,7 @@ def test_with_csvs(tmp_path):
     rewardsperlp, _ = calcrewards.calcRewards(stakes, V, SYM, R, OCEAN_avail)
 
     sum_ = sum(rewardsperlp[chainID].values())
-    assert sum_ == pytest.approx(OCEAN_avail, 0.01), sum_
+    assert sum_ == pytest.approx(OCEAN_avail, OCEAN_avail/1000), sum_
     csvs.saveRewardsperlpCsv(rewardsperlp, csv_dir, "OCEAN")
     rewardsperlp = None  # ensure not used later
 

--- a/util/test/test_end-to-end.py
+++ b/util/test/test_end-to-end.py
@@ -29,6 +29,8 @@ def test_without_csvs():
     rng = blockrange.BlockRange(st, fin, n)
 
     (V0, SYM0) = query.queryNftvolsAndSymbols(rng, chainID)
+    V = {chainID: V0}
+    SYM = {chainID: SYM0}
 
     vebals = query.queryVebalances(rng, chainID)
     allocs = query.queryAllocations(rng, chainID)
@@ -36,12 +38,7 @@ def test_without_csvs():
 
     R = {"OCEAN": 0.5, "H2O": 1.618}
 
-    V, SYM = (
-        {chainID: V0},
-        {chainID: SYM0},
-    )
-
-    OCEAN_avail = 1e-6
+    OCEAN_avail = 1e-4
 
     rewardsperlp, _ = calcrewards.calcRewards(stakes, V, SYM, R, OCEAN_avail)
 
@@ -88,11 +85,11 @@ def test_with_csvs(tmp_path):
 
     stakes = allocations.loadStakes(csv_dir)  # loads allocs & vebals, then *
 
-    OCEAN_avail = 1e-6
+    OCEAN_avail = 1e-4
     rewardsperlp, _ = calcrewards.calcRewards(stakes, V, SYM, R, OCEAN_avail)
 
     sum_ = sum(rewardsperlp[chainID].values())
-    assert sum_ == pytest.approx(OCEAN_avail, OCEAN_avail/1000), sum_
+    assert sum_ == pytest.approx(OCEAN_avail, OCEAN_avail / 1000), sum_
     csvs.saveRewardsperlpCsv(rewardsperlp, csv_dir, "OCEAN")
     rewardsperlp = None  # ensure not used later
 

--- a/util/test/test_end-to-end.py
+++ b/util/test/test_end-to-end.py
@@ -21,7 +21,6 @@ ST = 0
 FIN = 0
 
 
-@pytest.mark.skip(reason="Don't skip, once fixed #354")
 @enforce_types
 def test_without_csvs():
     chainID = networkutil.DEV_CHAINID
@@ -50,7 +49,6 @@ def test_without_csvs():
     assert sum_ == pytest.approx(OCEAN_avail, 0.01), sum_
 
 
-@pytest.mark.skip(reason="Don't skip, once fixed #354")
 @enforce_types
 def test_with_csvs(tmp_path):
     """

--- a/util/test/test_end-to-end.py
+++ b/util/test/test_end-to-end.py
@@ -46,7 +46,7 @@ def test_without_csvs():
     rewardsperlp, _ = calcrewards.calcRewards(stakes, V, SYM, R, OCEAN_avail)
 
     sum_ = sum(rewardsperlp[chainID].values())
-    assert sum_ == pytest.approx(OCEAN_avail, OCEAN_avail/1000.0), sum_
+    assert sum_ == pytest.approx(OCEAN_avail, OCEAN_avail / 1000.0), sum_
 
 
 @enforce_types

